### PR TITLE
[#75476527] Release pipeline

### PR DESCRIPTION
**Pull request by balanza**
Same changes applied in https://github.com/pagopa/io-functions-services/pull/96.

### Changes in brief
* Introduce the `Release` stage on the pipeline which marks a release on the repository
* Remove `release-it` scripts and use plain `npm version` instead
* Refactor the whole CI/CD to have 
  * separated pipelines for code review and deploy scenarios
  * everything is into `.devops` folder now

### CODE REVIEW

File: .devops/deploy-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        steps:  
+          - template: templates/rest-healthcheck/template.yaml@pagopaCommons 
+            parameters:
+              azureSubscription: '$(PRODUCTION_AZURE_SUBSCRIPTION)'
+              appName: '$(PRODUCTION_APP_NAME)'
+              endpoint: 'https://$(PRODUCTION_APP_NAME)-staging.azurewebsites.net/api/v1/info'
+              endpointType: 'private'
+              containerInstanceResourceGroup: 'io-p-rg-common'
+              containerInstanceVNet: 'io-p-vnet-common'
+              containerInstanceSubnet: 'azure-devops'
```
Comment (pasqualedevita) [inline .devops/deploy-pipelines.yml:151]: Why if fn-assets prod slot is public while staging slot has access restrictions?

prod slot
https://github.com/pagopa/io-infrastructure-live-new/blob/master/prod/westeurope/internal/api/functions_assets_r3/function_app/terragrunt.hcl#L107

staging slot
https://github.com/pagopa/io-infrastructure-live-new/blob/master/prod/westeurope/internal/api/functions_assets_r3/function_app_slot_staging/terragrunt.hcl#L108

for public endpoints we can set simply set
```suggestion
              endpointType: 'public'
```

File: .devops/deploy-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+      - Deploy_staging
+    jobs:
+      - job: 'do_healthcheck'       
+        steps:  
+          - template: templates/rest-healthcheck/template.yaml@pagopaCommons 
+            parameters:
+              azureSubscription: '$(PRODUCTION_AZURE_SUBSCRIPTION)'
+              appName: '$(PRODUCTION_APP_NAME)'
+              endpoint: 'https://$(PRODUCTION_APP_NAME)-staging.azurewebsites.net/api/v1/info'
+              endpointType: 'public'
```
Comment (pasqualedevita) [inline .devops/deploy-pipelines.yml:148]: setting to public needs to remove allowed subnets here

https://github.com/pagopa/io-infrastructure-live-new/blob/master/prod/westeurope/internal/api/functions_assets_r3/function_app_slot_staging/terragrunt.hcl#L108

File: .devops/deploy-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    dependsOn:
+      - Deploy_staging
+    jobs:
+      - job: 'do_healthcheck'       
+        steps:  
+          - template: templates/rest-healthcheck/template.yaml@pagopaCommons 
+            parameters:
+              azureSubscription: '$(PRODUCTION_AZURE_SUBSCRIPTION)'
+              appName: '$(PRODUCTION_APP_NAME)'
+              endpoint: 'https://$(PRODUCTION_APP_NAME)-staging.azurewebsites.net/api/v1/info'
```
Comment (pasqualedevita) [inline .devops/deploy-pipelines.yml:147]:
```suggestion
              endpoint: 'https://$(PRODUCTION_APP_NAME)-staging.azurewebsites.net/info'
```
https://github.com/pagopa/io-functions-assets/blob/master/Info/index.ts#L13

ps: this why is more useful have a standard endpoint

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

File: .devops/deploy-pipelines.yml (hunk 2 above)
Comment (balanza) [reply .devops/deploy-pipelines.yml:148]: What do we do?

File: .devops/deploy-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+      - Deploy_staging
+    jobs:
+      - job: 'do_healthcheck'       
+        steps:  
+          - template: templates/rest-healthcheck/template.yaml@pagopaCommons 
+            parameters:
+              azureSubscription: '$(PRODUCTION_AZURE_SUBSCRIPTION)'
+              appName: '$(PRODUCTION_APP_NAME)'
+              endpoint: 'https://$(PRODUCTION_APP_NAME)-staging.azurewebsites.net/info'
+              endpointType: 'public'
```
Comment (pasqualedevita) [inline .devops/deploy-pipelines.yml:152]:
```suggestion
              endpointType: 'private'
              containerInstanceResourceGroup: 'io-p-rg-common'
              containerInstanceVNet: 'io-p-vnet-common'
              containerInstanceSubnet: 'azure-devops' 
```
sorry rollback to private

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

**Review by pasqualedevita — APPROVED** [review]

**Pull request merged**